python-checks/src/test/java/org/sonar/python/checks/CognitiveComplexityFunctionCheckTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import org.junit.Test;
+import org.sonar.python.PythonAstScanner;
+import org.sonar.python.checks.utils.PythonCheckVerifier;
+import org.sonar.squidbridge.api.SourceFile;
+import org.sonar.squidbridge.checks.CheckMessagesVerifier;
+
+public class CognitiveComplexityFunctionCheckTest {
+
+  private final CognitiveComplexityFunctionCheck check = new CognitiveComplexityFunctionCheck();
+  private static final String FILE_PATH = "src/test/resources/checks/cognitiveComplexityFunction.py";
+
+  @Test
+  public void test() {
+    check.setThreshold(0);
+    PythonCheckVerifier.verify(FILE_PATH, check);
+  }
+
+  @Test
+  public void default_threshold() throws Exception {
+    SourceFile file = PythonAstScanner.scanSingleFile("src/test/resources/checks/fileComplexity.py", check);
+    CheckMessagesVerifier.verify(file.getCheckMessages())
+      .noMore();
```

Comment:
@vilchik-elena I would prefer to have a dedicated test file for the default threshold where we have one function exactly on the "max value" and one function which is the almost the same but has "max value + 1".